src/main/java/bio/terra/service/filedata/azure/tables/TableDirectoryDao.java

```diff
@@ -180,21 +180,20 @@ public FireStoreDirectoryEntry retrieveByPath(
       TableServiceClient tableServiceClient, List<String> refIdArray) {
     logger.info("validateRefIds for {} file ids", refIdArray.size());
     TableClient tableClient = tableServiceClient.getTableClient(TABLE_NAME);
-    return Lists.partition(refIdArray, MAX_FILTER_CLAUSES).stream()
+    return ListUtils.partition(refIdArray, MAX_FILTER_CLAUSES).stream()
         .flatMap(
             refIds -> {
-              var filter =
+              String filter =
                   refIds.stream()
                       .map(refId -> String.format("fileId eq '%s'", refId))
                       .collect(Collectors.joining(" or "));
               ListEntitiesOptions options = new ListEntitiesOptions().setFilter(filter);
               PagedIterable<TableEntity> entities = tableClient.listEntities(options, null, null);
-              var validRefIds =
+              List<String> validRefIds =
                   entities.stream()
                       .map(e -> e.getProperty("fileId").toString())
                       .collect(Collectors.toList());
-              refIds.removeAll(validRefIds);
-              return refIds.stream();
+              return ListUtils.subtract(refIds, validRefIds).stream();
```

Comment:
If you're returning a stream anyway, you can use `filter` here, since there's no need to compute the difference as a list in memory. Something like
```
return refIds.stream().filter(validRefIds::contains);
```